### ISSUE TITLE
Linter warnings on schema.org examples

### ISSUE THREAD

**Issue by gkellogg** [schemaorg/schemaorg#53]
Update: https://gist.github.com/danbri/c5fef76dcf89bc23bdb6 has a fresh run against 2015-01-28th snapshot of sdo-stantz branch. -- @danbri

FYI, the following are warnings produced by the Structured-Data Linter for current schema.org examples:

[ earlier results removed for readability, see link above ] 

This was done through the following:
- clone http://github.com/structured-data/linter
- bundle install
- run the schema_examples rake task to generate all example files
- run `script/parse schema.org/*.html --lint --quiet

Changes to the schema.org RDFS need to be synchronized by updating the vocabulary in the RDF.rb gem, typically done after each release.

**Comment by danbri** [comment]
Ok @gkellogg I finally got the linter running locally. Do you have a script handy that unpacks and checks the schema.org examples?

**Comment by scor** [comment]
@danbri side note: might be worth to run all the example through the new Google validator at https://developers.google.com/webmasters/structured-data/testing-tool/ - you might have a way to send all those examples in batch and get the warnings/errors back. Maybe you have already done so, in which case ignore this comment. Happy to create a new issue if not.

**Comment by danbri** [comment]
Yeah, we should do that! Thanks for the nudge, it's already on my todo list...

**Comment by gkellogg** [comment]
From the linter directory, you can run `rake schema:warnings` which will generate `etc/schema-warnings.txt` over the examples included in the linter. These examples can be updated using `rake schema:examples` which takes them from Guha's master branch, but could easily be modified to take them from somewhere else.

**Comment by danbri** [comment]
Thanks Gregg! Note that Guha's master branch has graduated to an organization account within github now. Most Web links seem to redirect nicely, hopefully it will continue to work for you too.

-> https://github.com/schemaorg/schemaorg

There's still some admin/workflow setup to do but at least the basic shape of things is settling down.

**Comment by danbri** [comment]
FYI the 'rake schema:examples' command wiped out the examples collection because sinatra-linkeddata failed (I think it wasn't installed properly)

**Comment by danbri** [comment]
It needed 

          gem install sinatra-linkeddata
          gem install active_support-custom_logger

... and then generates a load of (very useful) output, then appears to hang at schema.org/UpdateAction-163-jsonld.html, file is...  41665 Jan 19 22:14 etc/schema-warnings.txt

**Comment by gkellogg** [comment]
Thanks, I'll update the dependencies and check out the UpdateAction issue.

**Comment by gkellogg** [comment]
FYI, to run locally, run `bundle install`, then it will pull down all dependencies. I'll push up a change to the README.

Note that I have some fixes pending in PR #199 which you might want to pick up. I believe one of those fixes "solved" the JSON-LD hang.

If you do the `bundle install` and then `rake schema:warnings` it should run to completion.

**Comment by danbri** [comment]
I used "bundle install --path vendor/bundler" per the readme.

**Comment by danbri** [comment]
ping @RichardWallis ... re-assigning this to you, as I clearly didn't get to it!

**Comment by danbri** [comment]
Nearby,

 * https://twitter.com/danbri/status/823936803463987200
 * https://github.com/structured-data/linter/blob/master/.travis.yml

**Comment by danbri** [comment]
For desktop developer use I don't think it made sense to have a strong dependency on Ruby tooling, however now that we are generally [wired into Travis-CI](https://travis-ci.org/schemaorg/schemaorg) for our unit tests (thanks @lanthaler :), it is a  good time to revisit this. We are already set up to skip the rdflib-based tests if that library is not available, so we could do similarly for the ruby-powered checks. See https://github.com/schemaorg/schemaorg/blob/sdo-callisto/tests/test_graphs.py#L57

**Comment by gkellogg** [comment]
I'm on it.

**Comment by danbri** [comment]
https://travis-ci.org/schemaorg/schemaorg shows 141 markup errors. Let's try to get that down to respectable levels for our next release!

We should also try to make sure that the examples use no terms that have been supersededBy another or moved to the attic.schema.org area.

**Comment by danbri** [comment]
We have not much improved this situation. https://travis-ci.org/github/schemaorg/schemaorg/jobs/709160005 reports a bunch of issues, mix of vocabulary stuff and syntax. 

@RichardWallis - how about we work through them together?

**Comment by Nattaponmacks** [comment]
**Update:** https://gist.github.com/danbri/c5fef76dcf89bc23bdb6 has succeeded in a fresh run against the snapshot of the sdo-stands branch from January 28, 2015. — @danbri

**Note:** The following warnings were produced by the Structured-Data Linter for the current schema.org examples:

[Earlier results removed for readability; see the link above.]

This process was carried out through the following steps:

1. Clone the repository from http://github.com/structured-data/linter (https://github.com/structured-data/linter).
2. Run `bundle install`.
3. Execute the schema_examples rake task to generate all example files.
4. Use the command `script/parse schema.org/*.html --lint --quiet`.

Please note that any changes to the schema.org RDF Schema (RDFS) need to be synchronized by updating the vocabulary in the RDF.rb gem. This is typically done after each release.

https://github.com/schemaorg/schemaorg/issues/53#issue-36231832

**Comment by gkellogg** [comment]
Note that the linter is effectively part of the schema.org GitHub Action, which is run against the state of the vocabulary in the repository. It should produce similar results to running using the method above.

**Comment by fthobe** [comment]
@gmackenz can be closed as pointed out by @gkellogg [in the comment above].(https://github.com/schemaorg/schemaorg/issues/53#issuecomment-2445148104)
